src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs

```diff
@@ -125,26 +126,66 @@ public static string[] GetAudienceCandidatesFromLaunchSettings(string project)
                     var profilesEnumerator = profiles.EnumerateObject();
                     foreach (var profile in profilesEnumerator)
                     {
-                        if (profile.Value.TryGetProperty("commandName", out var commandName))
+                        if (ExtractKestrelUrlsFromProfile(profile) is { } kestrelUrls)
                         {
-                            if (commandName.ValueEquals("Project"))
-                            {
-                                if (profile.Value.TryGetProperty("applicationUrl", out var applicationUrl))
-                                {
-                                    var value = applicationUrl.GetString();
-                                    if (value is { } applicationUrls)
-                                    {
-                                        return applicationUrls.Split(';');
-                                    }
-                                }
-                            }
+                            applicationUrls.AddRange(kestrelUrls);
+                        }
+
+                        if (ExtractIISExpressUrlFromProfile(profile) is { } iisUrls)
+                        {
+                            applicationUrls.AddRange(iisUrls);
                         }
                     }
                 }
             }
         }
 
-        return null;
+        return applicationUrls;
+
+        static List<string> ExtractIISExpressUrlFromProfile(JsonProperty profile)
+        {
+
+            if (profile.NameEquals("iisSettings"))
+            {
+                if (profile.Value.TryGetProperty("iisExpress", out var iisExpress))
+                {
+                    List<string> iisUrls = new();
+                    if (iisExpress.TryGetProperty("applicationUrl", out var iisUrl))
+                    {
+                        iisUrls.Add(iisUrl.GetString());
```

Comment:
can these urls also be `;` separated?

Reply:
I didn't think so, but we'd need to check with VS